tests/components/pterodactyl/test_config_flow.py

```diff
@@ -105,3 +105,71 @@ async def test_service_already_configured(
 
     assert result["type"] is FlowResultType.ABORT
     assert result["reason"] == "already_configured"
+
+
+@pytest.mark.usefixtures("mock_pterodactyl", "mock_setup_entry")
+async def test_reauth_full_flow(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test reauth config flow success."""
+    mock_config_entry.add_to_hass(hass)
+    result = await mock_config_entry.start_reauth_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reauth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_API_KEY: TEST_API_KEY}
+    )
+    await hass.async_block_till_done()
```

Comment:
```suggestion
```

Reply:
Why is that required for the user step, but not for the reauth step?